setuptools/build_meta.py

```diff
@@ -246,33 +264,35 @@ class _BuildMetaLegacyBackend(_BuildMetaBackend):
     packaging mechanism,
     and will eventually be removed.
     """
-    def run_setup(self, setup_script='setup.py'):
+
+    def run_command(self, *args):
         # In order to maintain compatibility with scripts assuming that
         # the setup.py script is in a directory on the PYTHONPATH, inject
         # '' into sys.path. (pypa/setuptools#1642)
         sys_path = list(sys.path)           # Save the original path
 
-        script_dir = os.path.dirname(os.path.abspath(setup_script))
+        if not os.path.exists(SETUP_SCRIPT) or os.stat(SETUP_SCRIPT).st_size == 0:
```

Comment:
Same as above, it's best to have an explanatory var
```suggestion
        setup_py_is_empty = not os.path.exists(SETUP_SCRIPT) or os.stat(SETUP_SCRIPT).st_size == 0
        if setup_py_is_empty:
```